crates/ott-balancer/src/balancer.rs

```diff
@@ -191,7 +191,7 @@ pub struct BalancerContext {
     pub monoliths: HashMap<MonolithId, BalancerMonolith>,
     pub rooms_to_monoliths: HashMap<RoomName, RoomLocator>,
     pub monoliths_by_region: HashMap<String, Vec<MonolithId>>,
-    pub monolith_selection: Box<dyn MonolithSelection + Send + Sync + 'static>,
+    pub monolith_selection: Box<dyn MonolithSelection + 'static + Send + Sync>,
```

Comment:
`enum_dispatch` allows us to drop the `Box<dyn T>` and replace it with the enum type